aslam_cv_pipeline/src/visual-npipeline.cc

```diff
@@ -101,9 +100,13 @@ size_t VisualNPipeline::getNumFramesComplete() const {
 }
 
 std::shared_ptr<VisualNFrame> VisualNPipeline::getNext() {
+  std::unique_lock<std::mutex> lock(mutex_);
+  return getNextImpl();
+}
+
+std::shared_ptr<VisualNFrame> VisualNPipeline::getNextImpl() {
   // Initialize the return value as null
   std::shared_ptr<VisualNFrame> rval;
-  std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
huh, where did the mutex go?

Reply:
into the public function. The diff is quite misleading.

Reply:
ah yeah right, sorry. I was looking at this line as if it was the declaration/definition of the mutex_ in the class header, but it's obviously not. never mind. everything good.